fronts-client/src/components/video/VideoControls.tsx

```diff
@@ -224,47 +227,67 @@ export const VideoControls = ({
 		replacementVideoAtomProperties,
 	]);
 
+	useEffect(() => {
+		setReplacementVideoIsSelfHosted(
+			showReplacementVideo &&
+				replacementVideoAtom !== undefined &&
+				replacementVideoAtomProperties?.platform === 'url',
+		);
+	}, [
+		replacementVideoAtom,
+		showReplacementVideo,
+		replacementVideoAtomProperties,
+	]);
+
+	useEffect(() => {
+		setMainMediaIsSelfHosted(
+			showMainVideo &&
+				mainMediaVideoAtom !== undefined &&
+				mainMediaVideoAtomProperties?.platform === 'url',
+		);
+	}, [showMainVideo, mainMediaVideoAtom, mainMediaVideoAtomProperties]);
+
 	if (!showMainVideo && !showReplacementVideo) {
 		return null;
 	}
 
-	const replacementVideoControls = document.getElementById(
-		replacementVideoControlsId,
-	);
-
-	const warningsContainer = document.getElementById(warningsContainerId);
-
-	const mainMediaIsSelfHosted =
-		showMainVideo &&
-		mainMediaVideoAtom !== null &&
-		mainMediaVideoAtomProperties?.platform === 'url';
+	const extraVideoControls = document.getElementById(extraVideoControlsId);
 
-	const replacementVideoIsSelfHosted =
-		showReplacementVideo &&
-		replacementVideoAtom !== null &&
-		replacementVideoAtomProperties?.platform === 'url';
+	const enableLoopingVideoFeatureSwitch =
+		pageConfig?.userData?.featureSwitches.find(
+			(feature) => feature.key === 'enable-looping-video',
+		);
 
 	return (
 		<>
+			{extraVideoControls !== null &&
+			(mainMediaIsSelfHosted || replacementVideoIsSelfHosted) &&
+			enableLoopingVideoFeatureSwitch?.enabled === true
```

Comment:
Does ` === true` need to be here? This might be a style choice in this repo but I thought I'd mention it